source/enemies/earthlyknight.js

```diff
@@ -0,0 +1,80 @@
+const { EnemyTemplate } = require("../classes/index.js");
+const { dealDamage, removeModifier, changeStagger } = require("../util/combatantUtil.js");
+const { isBuff } = require("../modifiers/_modifierDictionary");
+const { selectRandomFoe, selectNone, selectAllFoes } = require("../shared/actionComponents.js");
+const { spawnEnemy } = require("../util/roomUtil.js");
+const { getEmoji } = require("../util/elementUtil.js");
+
+const asteroid = require("./asteroid.js")
+
+const PATTERN = {
+  "Call Asteroid": "Tremor Smash",
+  "Damping Wallop": "random",
+  "Tremor Smash": "random"
+}
+function earthlyKnightPattern(actionName) {
+  return PATTERN[actionName]
+}
+
+module.exports = new EnemyTemplate("Earthly Knight",
+  "Earth",
+  250,
+  100,
+  "6",
+  0,
+  "Tremor Smash",
+  false
+).addAction({
+  name: "Damping Wallop",
+  element: "Earth",
+  description: `Inflict ${getEmoji("Earth")} damage and remove a buff`,
+  priority: 0,
+  effect: (targets, user, isCrit, adventure) => {
+    let damage = user.getPower() + 75;
+    changeStagger(targets, "elementMatchFoe");
+    let resultString = ""
+    for (target of targets) {
```

Comment:
```suggestion
    for (const target of targets) {
```
I've you're sticking with a `for(... of ...) loop (which I recommend against in a further down comment), don't forget to add a declarative to the iterator variable (`target`) so it doesn't get stuck in global scope.